tests/numerics/diagonal_matrix_test.C

```diff
@@ -0,0 +1,222 @@
+// libmesh includes
+#include <libmesh/diagonal_matrix.h>
+#include <libmesh/parallel.h>
+#include <libmesh/auto_ptr.h>
+#include <libmesh/dense_matrix.h>
+#include <libmesh/id_types.h>
+#include <libmesh/numeric_vector.h>
+
+#include <libmesh/petsc_vector.h>
+#include <petsc.h>
+
+#include "libmesh_cppunit.h"
+#include "test_comm.h"
+
+#include <memory>
+#include <numeric>
+
+#define UNUSED 0
+
+using namespace libMesh;
+
+namespace
+{
+template <typename T>
+T
+termial(T n)
+{
+  if (n == 1)
+    return 1;
+  return n + termial(n - 1);
```

Comment:
Is this just a really expensive way to calculate ```n*(n+1)/2```?

Reply:
Lol. Will change

Reply:
Thanks, I just learned a [new math word/notation](https://en.wikipedia.org/wiki/Termial)!